### PR TITLE
Add support for 'file' button type

### PR DESCRIPTION
**Pull request by ryanjbonnell**

### CODE REVIEW

**Comment by jerone** [comment]
Replaces https://github.com/cakecatz/flex-toolbar/pull/71
Fixes https://github.com/cakecatz/flex-toolbar/issues/47

**Comment by jerone** [comment]
LGTM! 

Don't have an environment to test; @cakecatz you have time?

**Comment by ryanjbonnell** [comment]
I used the following commands to aide in running a simple, manual test of the changes:

```shell
cd ~/.atom/dev

git clone git@github.com:ryanjbonnell/flex-toolbar.git flex-tool-bar && cd "$_"

git checkout feature/button-file

apm install && apm link --dev && atom --dev
```

I then used the following in my `toolbar.cson` configuration:

```javascript
{
  type: "file"
  icon: "file-code"
  url: "/Users/username/.atom/styles.less"
  tooltip: "Open Stylesheet..."
  style: {
    color: "red"
  }
  mode: "dev"
}
```

Everything worked in my local Atom dev environment, but admittedly your testing efforts will be far more comprehensive and automated than mine :)

**Comment by cakecatz** [comment]
@jerone :ok_woman: 

@ryanjbonnell 
This is worked perfectly!!
Good job :+1: 

Can you check on Windows? (I don't have Windows :sob: )

**Comment by UziTech** [comment]
@ryanjbonnell This works fine in windows. If you can rebase and update the readme to fix the merge conflict I will merge this

**Comment by ryanjbonnell** [comment]
@UziTech Great! I resolved the conflicts in the README and committed the changes for your review and testing.

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ type `Flex Tool Bar: Edit Config File` in the Atom command palette.
 ## Configuration
 
 **Flex Tool Bar** has four `type`s you can configure:
```
Comment (UziTech) [inline README.md]: this should be "five" instead  of "four"

File: types/file.coffee [hunk 2]
```diff
@@ -0,0 +1,13 @@
+UrlReplace = require '../lib/url-replace'
+
+module.exports = (toolBar, button) ->
+  return toolBar.addButton
+    icon: button.icon
+    callback: (url) ->
+      urlReplace = new UrlReplace()
+      url = urlReplace.replace url
+      atom.workspace.open (url)
```
Comment (UziTech) [inline types/file.coffee]: Is there a reason we are using UrlReplace here?

It seems unlikely that someone would want to use the name of the repo, owner, or atom version in a file path.

Comment (UziTech) [reply types/file.coffee]: Also does this handle relative paths?

Comment (UziTech) [reply types/file.coffee]: looks like it does handle relative paths pretty well

**Comment by UziTech** [comment]
I added this to the v1.0 release but I changed the syntax to use button.file instead of button.url

**Pull request closed without merging**